pkg/binder/utils/binder_common_state.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2024 The Godel Scheduler Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"fmt"
+
+	framework "github.com/kubewharf/godel-scheduler/pkg/framework/api"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	BinderCommonStateKey = "BinderCommonState"
+)
+
+type CommonState struct {
```

Comment:
We cannot write `victimsGroupsByNode` directly in `cycleState` because `cycleState` **requires** the data structure to implement the `Clone` function. But `victimsGroupsByNode` is just a `map` and does not implement the `Clone` function.

Reply:
You can define a struct that directly corresponds to `victimsGroupsByNode` instead of encapsulating it in `CommonState`.

Reply:
Now I have added a `PodGroupByNode` for `victimsGroupsByNode`. It implements the `Clone` function and can be directly read and written by `cycleState`.

Reply:
I have moved these `cycleState` related functions to `framework/api/common_state.go` so that it is more in line with the specifications of previous codes.